libs/spark/src/Unstable_Button/Unstable_Button.tsx

```diff
@@ -0,0 +1,244 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import {
+  default as MuiButton,
+  ButtonProps as MuiButtonProps,
+} from '@material-ui/core/Button';
+import makeStyles from '../makeStyles';
+import { OverridableComponent, OverrideProps } from '../utils';
+import { buildVariant } from '../theme/typography';
+import { lighten, darken } from '@material-ui/core/styles';
+
+export interface Unstable_ButtonTypeMap<
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  P = {},
+  D extends React.ElementType = 'button'
+> {
+  props: P &
+    Omit<MuiButtonProps, 'classes' | 'variant'> & {
+      /**
+       * The variant to use.
+       */
+      variant?: 'primary' | 'stroked' | 'ghost' | 'destructive';
+    };
+  defaultComponent: D;
+  classKey: Unstable_ButtonClassKey;
+}
+
+export type Unstable_ButtonProps<
+  D extends React.ElementType = Unstable_ButtonTypeMap['defaultComponent'],
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  P = {}
+> = OverrideProps<Unstable_ButtonTypeMap<P, D>, D>;
+
+export type Unstable_ButtonClassKey = 'root' | 'startIcon' | 'endIcon';
+
+// Extracted as we dont have a unstable_variant
+const buttonFontVariantSmall = buildVariant(
+  500,
+  14,
+  16,
+  undefined,
+  'none',
+  '"Inter", sans-serif',
+  "'cv05' 1, 'ss03' 1"
+);
+const buttonFontVariantMedium = buildVariant(
+  500,
+  16,
+  24,
+  undefined,
+  'none',
+  '"Inter", sans-serif',
+  "'cv05' 1, 'ss03' 1"
+);
+const buttonFontVariantLarge = buildVariant(
+  500,
+  18,
+  24,
+  undefined,
+  'none',
+  '"Inter", sans-serif',
+  "'cv05' 1, 'ss03' 1"
+);
+const useStyles = makeStyles<Unstable_ButtonClassKey>(
+  (theme) => ({
+    root: (props: Unstable_ButtonProps) => ({
+      borderRadius: 4,
+      ...(props.variant === 'primary' && {
+        backgroundColor: theme.unstable_palette.brand.blue,
+        color: theme.palette.common.white,
```

Comment:
Text color and font size are specific to the dom tag with the `label` class key, not `root`.